src/lib/uTop.cppo.ml

```diff
@@ -624,6 +630,37 @@ let () =
     (Toploop.Directive_none
        (fun () -> print_endline (Sys.getcwd ())))
 
+let () =
+  Hashtbl.add Toploop.directive_table "utop_stash"
+    (Toploop.Directive_string
+      (fun fname ->
+        let _ :: entries = LTerm_history.contents stashable_session_history in
+        (* getting and then reversing the entries instead of using
+         * [LTerm_history.save] because the latter has to be integrated
+         * into an LWT program flow. *)
```

Comment:
This is not a good reason as you can use `Lwt_main.run` here. Other directives are calling it. But I imagine you don't want to escape newlines as `LTerm_history.save` does. Can you update the comment?